jetty-websocket/websocket-javax-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/JavaxClientShutdownWithServerTest.java

```diff
@@ -0,0 +1,150 @@
+        contextHandler.addServlet(new ServletHolder(new ServletContainerInitializerShutdownServlet()), "/shutdownContainer");
+        server.setHandler(contextHandler);
+
+        // Because we are using embedded we must manually add the Javax WS Client Shutdown SCI.
+        JavaxWebSocketShutdownContainer javaxWebSocketClientShutdown = new JavaxWebSocketShutdownContainer();
+        shutdownContainer = javaxWebSocketClientShutdown;
+        contextHandler.addServletContainerInitializer(javaxWebSocketClientShutdown);
+
+        server.start();
+        serverUri = WSURI.toWebsocket(server.getURI());
+
+        httpClient = new HttpClient();
+        httpClient.start();
+    }
+
+    @AfterEach
+    public void after() throws Exception
+    {
+        httpClient.stop();
+        server.stop();
+    }
+
+    @Test
+    public void testShutdownWithContextHandler() throws Exception
+    {
+        ContentResponse response = httpClient.GET(serverUri.resolve("/contextHandler"));
+        assertThat(response.getStatus(), is(HttpStatus.OK_200));
+
+        assertNotNull(container);
+        assertThat(container, instanceOf(JavaxWebSocketClientContainer.class));
+        JavaxWebSocketClientContainer clientContainer = (JavaxWebSocketClientContainer)container;
+        assertThat(clientContainer.isRunning(), is(true));
+
+        // The container should be a bean on the ContextHandler.
+        Collection<WebSocketContainer> containedBeans = contextHandler.getBeans(WebSocketContainer.class);
+        assertThat(containedBeans.size(), is(1));
+        assertThat(containedBeans.toArray()[0], is(container));
+
+        // The client should be attached to the servers LifeCycle and should stop with it.
+        server.stop();
+        assertThat(clientContainer.isRunning(), is(false));
+        assertThat(server.getContainedBeans(WebSocketContainer.class), empty());
+    }
+
+    @Test
+    public void testShutdownWithShutdownContainer() throws Exception
+    {
+        ContentResponse response = httpClient.GET(serverUri.resolve("/shutdownContainer"));
+        assertThat(response.getStatus(), is(HttpStatus.OK_200));
+
+        assertNotNull(container);
+        assertThat(container, instanceOf(JavaxWebSocketClientContainer.class));
+        JavaxWebSocketClientContainer clientContainer = (JavaxWebSocketClientContainer)container;
+        assertThat(clientContainer.isRunning(), is(true));
+
+        // The container should be a bean on the ContextHandler.
+        Collection<WebSocketContainer> containedBeans = shutdownContainer.getBeans(WebSocketContainer.class);
+        assertThat(containedBeans.size(), is(1));
+        assertThat(containedBeans.toArray()[0], is(container));
+
+        // The client should be attached to the servers LifeCycle and should stop with it.
+        server.stop();
+        assertThat(clientContainer.isRunning(), is(false));
+        assertThat(server.getContainedBeans(WebSocketContainer.class), empty());
+    }
```

Comment:
I would add two tests that prove the behavior in case of multiple web applications, since we want to make sure that the static field `SHUTDOWN_CONTAINER` is not overwritten by other instances.

One test should have the classes provided by the server, and one by the web applications (both).